packages/amplify-category-function/provider-utils/awscloudformation/utils/cron.js

```diff
@@ -0,0 +1,267 @@
+async function askScheduleRuleQuestions(context, resourceName, parameters) {
+  const projectBackendDirPath = context.amplify.pathManager.getBackendDirPath();
+  const resourceDirPath = path.join(projectBackendDirPath, categoryName, resourceName);
+  const cfnFileName = `${resourceName}-cloudformation-template.json`;
+  const cfnFilePath = path.join(resourceDirPath, cfnFileName);
+
+  if (!parameters || !parameters.cloudwatchEvent || parameters.cloudwatchEvent === 'NONE') {
+    if (await context.amplify.confirmPrompt.run('Do you want to schedule this lambda function?', false)) {
+      parameters.cloudwatchEvent = 'true';
+      try {
+        cloudWatchRule = await cronServiceWalkthrough(context, parameters);
+        parameters.cloudwatchRule = cloudWatchRule;
+        if (context.input.command === 'update') {
+          //append cloudwatch events to CFN File
+          const cfnContent = context.amplify.readJsonFile(cfnFilePath);
+          cfnContent.Resources.CloudWatchEvent = {
+            Type: 'AWS::Events::Rule',
+            Properties: {
+              Description: 'Schedule rule for Lambda',
+              ScheduleExpression: cloudWatchRule,
+              State: 'ENABLED',
+              Targets: [
+                {
+                  Arn: { 'Fn::GetAtt': ['LambdaFunction', 'Arn'] },
+                  Id: {
+                    Ref: 'LambdaFunction',
+                  },
+                },
+              ],
+            },
+          };
+          // append permissions to invoke lambda via cloiudwatch to CFN file
+          cfnContent.Resources.PermissionForEventsToInvokeLambda = {
+            Type: 'AWS::Lambda::Permission',
+            Properties: {
+              FunctionName: {
+                Ref: 'LambdaFunction',
+              },
+              Action: 'lambda:InvokeFunction',
+              Principal: 'events.amazonaws.com',
+              SourceArn: { 'Fn::GetAtt': ['CloudwatchEvent', 'Arn'] },
+            },
+          };
+          fs.writeFileSync(cfnFilePath, JSON.stringify(cfnContent, null, 4));
+        }
+      } catch (e) {
+        context.print.error(e.message);
+      }
+    }
+  } else {
+    if (await context.amplify.confirmPrompt.run('Do you want to Update/Remove the ScheduleEvent Rule?', false)) {
+      const cfnContent = context.amplify.readJsonFile(cfnFilePath);
+      const scheduleEventOperationQuestion = {
+        type: 'list',
+        name: 'ScheduleEventOperation',
+        message: 'Select from the following options',
+        choices: ['Update the CronJob', 'Remove the CronJob'],
+      };
+
+      const scheduleEventOperationAnswer = await inquirer.prompt([scheduleEventOperationQuestion]);
+
+      switch (scheduleEventOperationAnswer.ScheduleEventOperation) {
+        case 'Update the CronJob': {
+          // add service walkthrough to get the cron expression
+          cloudWatchRule = await cronServiceWalkthrough(context, parameters);
+          parameters.cloudwatchEvent = 'true';
```

Comment:
If we store a boolean in that parameter, then why do we've NONE as the default?

Reply:
fixed